src/key.rs

```diff
@@ -702,6 +719,22 @@ impl Connection {
                 message: format!("{:?}", Error::YubiKey(e)),
             }));
         }
+
+        #[cfg(target_os = "linux")]
+        if let Ok(ring) = maybe_ring {
+            ring.add_key(&description, &pin.expose_secret().as_bytes())
+                .map_err(|e| format!("add_key({description}, pin): {e}"))
+                .and_then(|key| {
+                    key.set_timeout(300).map_err(|e| format!("set_timeout(300): {e}"))?;
```

Comment:
- How reliable is this timeout?
- Given that you are not setting a timeout at the point where the PIN is added to the keyring (I presume due to limitations with the `linux-keyutils` API), this means that if the keyring addition succeeds and then the timeout-setting fails, the PIN is now (IIUC) permanently stored in the keyring (which is not acceptable behaviour).